### DISCUSSION TITLE
Touchscreen support in the upstream kernel

### DISCUSSION THREAD

**Discussion by ruslanbay** [linux-surface/linux-surface#796] [General]
First, thank you so much for you work! I'm using Fedora with default kernel as my main OS since [5.12](https://github.com/linux-surface/linux-surface/issues/96#issuecomment-826698672). I don't care that much about camera, but I'm really missing touchscreen support. Do you have plans to push this feature to the upstream?

**Comment by qzed** [comment]
I'd like to get it upstream at some point, but I think there's still a bit of work left to do and the userspace interface may change. When we push it upstream we'll have to commit to a fixed userspace interface, so it has to be solid.
